main.tf

```diff
@@ -0,0 +1,45 @@
+terraform {
+  required_version = ">= 0.11.11"
+}
+
+# Create ECS service autoscaling policies
+module "autoscale_service" {
+  source = "./modules/service-autoscaling"
+
+  cluster_name = "${var.cluster_name}"
+  service_name = "${var.service_name}"
+  queue_name   = "${var.queue_name}"
+
+  max_capacity = "${var.service_max_capacity}"
+  min_capacity = "${var.service_min_capacity}"
+
+  target_value = "${var.queue_backlog_target_value}"
+}
+
+data "aws_lambda_function" "compute_queue_backlog" {
+  function_name = "${var.lambda_name}"
+}
+
+# Create cloudwatch event resources to invoke the compute-queue-backlog lambda
+resource "aws_cloudwatch_event_rule" "compute_queue_backlog" {
+  name                = "${data.aws_lambda_function.compute_queue_backlog.function_name}-${var.service_name}"
+  description         = "Schedule execution of ${data.aws_lambda_function.compute_queue_backlog.function_name} to compute queue backlog metrics for ${var.service_name} ${var.queue_name} queue."
+  schedule_expression = "${var.lambda_invocation_interval}"
+}
+
+resource "aws_cloudwatch_event_target" "compute_queue_backlog" {
+  rule      = "${aws_cloudwatch_event_rule.compute_queue_backlog.name}"
+  target_id = "${data.aws_lambda_function.compute_queue_backlog.function_name}-${var.service_name}"
+  arn       = "${data.aws_lambda_function.compute_queue_backlog.arn}"
+  input     = "{\"cluster_name\": \"${var.cluster_name}\", \"service_name\": \"${var.service_name}\", \"queue_name\": \"${var.queue_name}\", \"queue_owner_aws_account_id\": \"${var.queue_owner_aws_account_id != "" ? var.queue_owner_aws_account_id : data.aws_caller_identity.current.account_id}\", \"est_msgs_per_sec\": \"${var.service_est_msgs_per_sec}\"}"
+}
+
+resource "aws_lambda_permission" "compute_queue_backlog" {
+  statement_id  = "AllowExecutionFromCloudWatch"
+  action        = "lambda:InvokeFunction"
+  function_name = "${data.aws_lambda_function.compute_queue_backlog.function_name}"
+  principal     = "events.amazonaws.com"
+  source_arn    = "${aws_cloudwatch_event_rule.compute_queue_backlog.arn}"
+}
+
+data "aws_caller_identity" "current" {}
```

Comment:
What if the queue owner isn't the same as the executing account? This is the case with felix.

Reply:
Systems that need to grant the lambda access to a queue in a different account will need to add the permissions, like we do for felix. I could add the permissions, like I do for cloudwatch rules; however, the situation is different because the queue resource is not something this module creates, so I do not want to start adding policy documents to resources that are largely unknown to the module. Thoughts?

Reply:
I agree - trying to think about the best way to handle this that doesn't involve creating permissions that won't be used. Maybe having a variable for `account_id` and using the current calling identity if none is specified?

Reply:
That's what I do. `queue_owner_aws_account_id`

Reply:
Got it  - missed that.